File: 3612-process-string-with-special-operations-i/3612-process-string-with-special-operations-i.java
class Solution {
    public String processStr(String s) {
        StringBuilder sb = new StringBuilder();

        for (char c: s.toCharArray()) {
            if (Character.isLowerCase(c)){
                sb.append(c);
            }
            if (c == '*') {
                if (!sb.isEmpty()) {
                    sb.setLength(sb.length() - 1);
                }
            }
            if(c == '#'){
                sb.append(sb);
            } 
            if(c == '%'){
                sb.reverse();
            } 
        }

        return sb.toString();
    }
}

File: 3612-process-string-with-special-operations-i/3612-process-string-with-special-operations-i.cpp
class Solution {
public:
    string processStr(string s) {
        string res = "";

        for (char c: s) {
            if (islower(c) != 0){
                res += c;
            }
            if (c == '*') {
                if (!res.empty()) {
                    res.pop_back();
                }
            }
            if(c == '#'){
                res += res;
            } 
            if(c == '%'){
                reverse(res.begin(), res.end());
            } 
        }

        return res;
    }
};